### PR TITLE
Make docker work

### PR DESCRIPTION
**Pull request by rolandtritsch**
## What/Why?

* Added a section on how to build and run a/the billmonger docker container
* To make this work I had to make sure we can configure the output and assets dirs

### CODE REVIEW

File: billings/billing.example.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ business:
  name:       "My Software Company"
  person:     "Billie Bob McAwesome"
  address:    "1234 Awesome Street, Funny Town, Postal 16, Ireland"
- image_file: "assets/MySoftware.png"
+ image_file: "MySoftware.png"
```
Comment (relistan) [inline billings/billing.example.yaml]:
```suggestion
 image_file: "assets/MySoftware.png"
```

File: README.md [hunk 2]
```diff
@@ -67,13 +67,28 @@ You may ask for help on the command line in the semi-standard way:
...
-  -b, --billing-date="2019-03-06"
-              The date to assume the bill is written on
+                        The YAML config file to use
+  -b, --billing-date="2019-12-29"
+                        The date to assume the bill is written on
+  -o, --output-dir="."  The output directory to use
+  -a, --assets-dir="."  The assets directory to use
 ```
+
+Using docker
```
Comment (relistan) [inline README.md]:
```suggestion
Using Docker
```

File: README.md [hunk 3]
```diff
@@ -67,13 +67,28 @@ You may ask for help on the command line in the semi-standard way:
...
+------------
+
+To build it and run it with docker you need to ...
```
Comment (relistan) [inline README.md]:
```suggestion
To build it and run it directly from a Docker container you need to:
```

File: .gitignore [hunk 4]
```diff
@@ -1,5 +1,4 @@
 *.swp
 *.swo
 vendor/
-billing.yaml
-
+billings/*.pdf
```
Comment (relistan) [inline .gitignore]: billing.yaml

Comment (relistan) [reply .gitignore]:
```suggestion
billings/*.pdf
billing.yaml
```

**Comment by relistan** [comment]
Thanks @rolandtritsch ! Only thing is we can rename `billings` -> `invoices`?

File: billings/billing.example.yaml (hunk 1 above)
Comment (rolandtritsch) [reply billings/billing.example.yaml]: Fixed.

File: README.md (hunk 3 above)
Comment (rolandtritsch) [reply README.md]: Fixed.

File: .gitignore (hunk 4 above)
Comment (rolandtritsch) [reply .gitignore]: Fixed.

**Comment by rolandtritsch** [comment]
> Thanks @rolandtritsch ! Only thing is we can rename `billings` -> `invoices`?

Done.

File: README.md [hunk 5]
```diff
@@ -67,13 +67,28 @@ You may ask for help on the command line in the semi-standard way:
...
+                        The YAML config file to use
+  -b, --billing-date="2019-12-29"
+                        The date to assume the bill is written on
+  -o, --output-dir="."  The output directory to use
+  -a, --assets-dir="."  The assets directory to use
 ```
+
+Using Docker
+------------
+
```
Comment (relistan) [inline README.md]:
```suggestion
You can run Billmonger from the command line. But some folks have
found that it's useful to run it from a Docker container directly.
This is also supported! Here are the steps in order to do that.
```

File: README.md [hunk 6]
```diff
@@ -67,13 +67,28 @@ You may ask for help on the command line in the semi-standard way:
...
+To build it and run it directly from a Docker container you need to:
+
+* make the changes and then ...
```
Comment (relistan) [inline README.md]:
```suggestion
The following assumes that you have a directory in your current path
called `./billmonger/invoices` where you would like to output PDF files
from Billmonger. You can substitute this for any other local path 
that is convenient.

We will also need to be able to mount the config file from our local 
filesystem into the Docker container. In the example below this file
also lives in `./billmonger/invoices`

Similarly, we will want to mount assents like the logo files from a
local directory. This is assumed to be `./billmonger/assets` in  this
example.

```

File: README.md [hunk 7]
```diff
@@ -67,13 +67,28 @@ You may ask for help on the command line in the semi-standard way:
...
+* run `docker build . --tag billmonger` and then ...
+* run `docker run --volume ${PWD}/billmonger/invoices:/invoices --volume ${PWD}/billmonger/assets:/assets billmonger --output-dir /invoices --assets-dir /assets --config-file /invoices/billing.example.yaml`
+
+Note: The config file needs to be in a mounted dir (e.g. invoices).
```
Comment (relistan) [inline README.md:106]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -67,13 +67,28 @@ You may ask for help on the command line in the semi-standard way:
...
+
+Note: To use the container in your environment you need to mount/configure the invoices and assets dirs and you probably want to alias the docker run command (e.g. `alias='docker run --volume ${PWD}/billmonger/invoices:/invoices --volume ${PWD}/billmonger/assets:/assets billmonger --output-dir /invoices --assets-dir /assets'`. With that you can run `billmonger --config-file /invoices/billing.example.yaml`
```
Comment (relistan) [inline README.md]:
```suggestion
Note: If you plan to use this a lot, you may want to alias the docker
run command with something like: 
`alias='docker run --volume ${PWD}/billmonger/invoices:/invoices --volume ${PWD}/billmonger/assets:/assets billmonger --output-dir /invoices --assets-dir /assets'`. 

You can then simply run `billmonger --config-file /invoices/billing.example.yaml`.
Make sure to _also_ run this from the correct relative path if you want
to have the directories mounted properly.
```

File: invoice/bill.go [hunk 9]
```diff
@@ -156,7 +156,7 @@ func (b *Bill) RenderToFile() error {
 	// It's safe to MustParse here because we validate args earlier
 	billTime := now.New(now.MustParse(b.config.Bill.Date))
 
-	outFileName := b.config.Business.Person + " " +
+	outFileName := b.config.OutputDir + "/" + b.config.Business.Person + "-" +
```
Comment (relistan) [inline invoice/bill.go]:
```suggestion
	outFileName := b.config.OutputDir + "/" + b.config.Business.Person + " " +
```
Don't want to change the filenaming.

File: invoice/config.go [hunk 10]
```diff
@@ -99,12 +99,14 @@ type BillingConfig struct {
 	Billables []BillableItem   `yaml:"billables"`
 	Bank      *BankDetails     `yaml:"bank"`
 	Colors    *BillColor       `yaml:"colors"`
+	OutputDir string
```
Comment (relistan) [inline invoice/config.go:102]:
```suggestion
	OutputDir string          `yaml:"output_dir"`
```

File: invoice/config.go [hunk 11]
```diff
@@ -99,12 +99,14 @@ type BillingConfig struct {
...
+	AssetsDir string
```
Comment (relistan) [inline invoice/config.go]:
```suggestion
	AssetsDir string          `yaml:"assets_dir"`
```

**Comment by relistan** [comment]
Thanks @rolandtritsch !

**Pull request merged**